bpf/bpf_host.c

```diff
@@ -1029,6 +1049,12 @@ int to_netdev(struct __ctx_buff *ctx __maybe_unused)
 	}
 #endif
 
+#if defined(ENABLE_EGRESS_GATEWAY)
+	ep_tail_call(ctx, CILIUM_CALL_IPV4_HANDLE_EGRESS_GW);
```

Comment:
I don't think that will work as we still want to execute the code below (BPF NodePort) if the packet doesn't match a connection requiring egress gateway SNATing. Since we don't come back from a tail call, that BPF NodePort code will be skipped.

Reply:
oh right, I should've add the if check here as well. how about just move `egress_nat_fwd` into tailcall? `snat_v4_process` is used in several places. I feel not all places are good for tailcalls (like in `tail_nodeport_nat_ipv4`)?

Reply:
I meant we could do a tail call from `handle_egress_nat_ipv4` to `tail_egress_nat_fwd` and the tail called program would just call `egress_nat_fwd`. We don't need to modify other places calling `egress_nat_fwd`. The `lookup_ip4_egress_endpoint` call needs to be before the tail call or we will skip BPF NodePort.